From 57da561c431eecc9de49635fef9123b26f2ed8b5 Mon Sep 17 00:00:00 2001
From: Jeremy Drake <github@jdrake.com>
Date: Thu, 7 Jan 2021 10:33:46 -0800
Subject: [PATCH] pkgconf: use configure/make instead of meson

This package is needed near the root of the dependency tree, long before
meson and ninja can be built.
---
 mingw-w64-pkgconf/PKGBUILD | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/mingw-w64-pkgconf/PKGBUILD b/mingw-w64-pkgconf/PKGBUILD
index 0c685e154..d346bf796 100644
--- a/mingw-w64-pkgconf/PKGBUILD
+++ b/mingw-w64-pkgconf/PKGBUILD
@@ -14,9 +14,7 @@ options=('!debug' 'strip')
 provides=("${MINGW_PACKAGE_PREFIX}-pkg-config")
 conflicts=("${MINGW_PACKAGE_PREFIX}-pkg-config")
 replaces=("${MINGW_PACKAGE_PREFIX}-pkg-config")
-makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
-             "${MINGW_PACKAGE_PREFIX}-meson"
-             "${MINGW_PACKAGE_PREFIX}-ninja")
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc")
 groups=("${MINGW_PACKAGE_PREFIX}-toolchain")
 source=("https://github.com/pkgconf/pkgconf/archive/refs/tags/$_realname-$pkgver.tar.gz"
         0001-fix-prefix-dirsep.patch
@@ -41,6 +39,7 @@ prepare() {
 
   patch -p1 -i ${srcdir}/0002-size-t-format.patch
   patch -p1 -i ${srcdir}/0003-printf-format.patch
+  autoreconf -fiv
 }
 
 build() {
@@ -48,20 +47,16 @@ build() {
   mkdir ${srcdir}/build-${MINGW_CHOST}
   cd ${srcdir}/build-${MINGW_CHOST}
 
-  MSYS2_ARG_CONV_EXCL="--prefix=" \
-  ${MINGW_PREFIX}/bin/meson \
-    --prefix="${MINGW_PREFIX}" \
-    --buildtype=plain \
-    -Dtests=false \
-    ../${_realname}-${pkgver}
+  ../${_realname}-${pkgver}/configure \
+    --prefix="${MINGW_PREFIX}"
 
-  meson compile
+  make
 }
 
 package() {
   cd ${srcdir}/build-${MINGW_CHOST}
 
-  DESTDIR="${pkgdir}" meson install
+  make install DESTDIR="${pkgdir}"
 
   cp "${pkgdir}${MINGW_PREFIX}"/bin/pkgconf "${pkgdir}${MINGW_PREFIX}"/bin/pkg-config
   cp "${pkgdir}${MINGW_PREFIX}"/bin/pkgconf "${pkgdir}${MINGW_PREFIX}"/bin/${MINGW_CHOST}-pkgconf
-- 
2.31.1.windows.1

